Upgrade Kibana to Elasticsearch 7.0

chrisdavies opened the pull request:
Upgrade Kibana to be compatible with ES 7.0.

Closes https://github.com/elastic/kibana/issues/23650

Each team is working on getting their own tests passing. That work should be merged here, and once this is green, we'll do a full code review on this and get it merged into master.

chrisdavies committed:
Make CI use the latest elasticsearch snapshot (#28854)

chrisdavies committed:
Merge branch 'upgrade/elasticsearch-7-0' of github.com:elastic/kibana into upgrade/elasticsearch-7-0

chrisdavies committed:
Upgrade es archives to support ES7.0 (#29179)

* Make migrations compatible with es 7
* Get past basic Kibana boot errors in saved object client
* Upgrade es-archives to support ES7

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/8514/)

chrisronline committed:
[Monitoring] Get monitoring tests working with mapping changes (#29188)

Delete es_archive mappings that are controlled by Elasticsearch index templates

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/8573/)

chrisdavies committed:
Fix uiSettings tests (#29258)

Merge master
Fix uiSettings tests

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/8784/)

chrisdavies committed:
Drop support for deleting the Kibana index (#29268)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/8802/)

chrisdavies committed:
Upgrade/es 7 tests (#29270)

Fix saved object tests

chrisdavies committed:
Upgrade es 7 tests (#29273)

Change doc -> _doc in context size functional tests

chrisdavies committed:
Merge master (#29274)

chrisdavies committed:
Merge branch 'master' into upgrade/elasticsearch-7-0

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/8814/)

chrisdavies committed:
Upgrade es 7 tests (#29276)

Make discover field_data sort test only the timestamp rather than
the full doc, as ES7 has modified the doc in some way.

chrisdavies committed:
Fix server mappings tests

chrisdavies committed:
Fix management functional tests

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/8825/)

chrisdavies committed:
Fix api_integration es archives

chrisdavies committed:
Fix management functional tests

chrisdavies committed:
Remove x-pack monitoring archive mappings

chrisdavies committed:
Merge branch 'master' of github.com:elastic/kibana into upgrade/elasticsearch-7-0

kobelb committed:
Fixing security functional tests for ES 7 (#29229)

chrisdavies committed:
Merge branch 'upgrade/elasticsearch-7-0' of github.com:elastic/kibana into upgrade/elasticsearch-7-0

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/8993/)

chrisdavies committed:
Temporary workaround for esjs and es 7 incompatibility

chrisdavies committed:
Merge branch 'master' of github.com:elastic/kibana into upgrade/elasticsearch-7-0

chrisdavies committed:
Fix logstash elasticsearch 7 tests

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/9016/)

chrisdavies committed:
Fix flakey alert test

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/9039/)

chrisdavies committed:
Fix BWC reporting tests

kobelb committed:
[6.x] Add new `allow_restricted_indices` field to the expected roles response result. (#29057) (#29350)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/9066/)

chrisdavies committed:
Merge branch 'master' of github.com:elastic/kibana into upgrade/elasticsearch-7-0

chrisdavies committed:
Merge branch 'upgrade/elasticsearch-7-0' of github.com:elastic/kibana into upgrade/elasticsearch-7-0

chrisdavies committed:
Fix monitoring mappings_exists tests

chrisdavies committed:
Fix es_archiver pre-7.0 logic

chrisdavies committed:
Fix bwc 6_3 mappings

chrisdavies committed:
Fix es_archiver test to handle include_type_name

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/9111/)

chrisronline committed:
[Monitoring] Change null to 0s (#29380)

Fix monitoring aggregation tests

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/9130/)

chrisdavies committed:
Merge branch 'master' of github.com:elastic/kibana into upgrade/elasticsearch-7-0

chrisdavies committed:
Merge branch 'master' of github.com:elastic/kibana into upgrade/elasticsearch-7-0

chrisdavies committed:
Fix logstash test

chrisdavies committed:
Merge branch 'upgrade/elasticsearch-7-0' of github.com:elastic/kibana into upgrade/elasticsearch-7-0

chrisdavies committed:
Merge master

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/9306/)

chrisdavies committed:
Fix telemetry tests

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/9325/)

chrisdavies committed:
Fix telemetry tests, maybe?

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/9346/)

jbudz commented on x-pack/test/api_integration/apis/xpack_main/telemetry/telemetry_local.js:
```diff
@@ -132,6 +57,8 @@ export default function ({ getService }) {
         .send({ timeRange })
         .expect(200);
 
+      // console.log(JSON.stringify(body, null, 2));
```
boop

chrisdavies replied:
Yeesh.

jbudz commented on src/server/saved_objects/service/create_saved_objects_service.js:
```diff
@@ -31,60 +31,18 @@ import { SavedObjectsClient } from './saved_objects_client';
 // Eventually, we hope to build a first-class import / export API, at which point, we can
 // remove the migrator from the saved objects client and leave only document validation here.
 export function createSavedObjectsService(server, schema, serializer, migrator) {
-  const onBeforeWrite = async () => {
```
is this meant to be removed in this PR?

chrisdavies replied:
Yeah. It was breaking a number of tests, so I figured: I can either fix all of those tests to work with a thing that we want to remove, or I can just remove it and do it all in this PR.

chrisdavies committed:
Remove console.log comment

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/9464/)

jbudz commented:
1) no template push before document update
1) removal of type in mappings in some cases, in other cases adding the query param to include type.
1) esarchiver support for both
1) if type still included type set to _doc

chrisdavies committed:
Merge master

Stacey-Gammon commented on test/functional/apps/discover/_field_data.js:
```diff
@@ -82,78 +82,7 @@ export default function ({ getService, getPageObjects }) {
       it('doc view should sort ascending', async function () {
 
         // Note: Could just check the timestamp, but might as well check that the whole doc is as expected.
```
I think can get rid of this comment now

chrisdavies replied:
Yup!

chrisdavies committed:
Remove superfluous (and misleading) comment

stacey-gammon reviewed (approved):
🎉 🎉 🎉

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/9730/)
